README.md

```diff
@@ -1,2 +1,135 @@
-# ros-pycharm-example
-A simple example for fully integrated Python development of a ROS / ROS2 project in PyCharm 
+# Developing Python ROS in PyCharm
+
+This is an example project to illustrate how use PyCharm as a fully integrated development environment for a ROS Python
+project by encapsulating the dependencies in a docker container.
+
+## The turtle example
+
+This example installs dependencies for the [`turtlesim`](http://wiki.ros.org/turtlesim) package on top of the base ROS
+image. It defines a new package called [`turtle_example`](./turtle_example) with a simple Python node
+called [`circler`](./turtle_example/scripts/circler)
+which publishes twist commands for a running turtlesim node to drive it in a circle.
+
+## Base ROS Docker images
+
+The majority of the background work is done in the
+repository [aica-technology/docker-images](https://github.com/aica-technology/docker-images). There, docker images are
+defined for specific versions of ROS and ROS2 that contain all of the necessary development dependencies, including an
+inbuilt SSH server. PyCharm will use the SSH server to access the remote environment.
+
+To begin, clone the [docker-images](https://github.com/aica-technology/docker-images) repository and follow the
+instructions to pull or build your base ROS workspace of choice.
+
+## PyCharm setup
+
+Prerequisites: Docker, PyCharm Professional 2021 or newer, base ROS docker image from AICA
+
+Step 1: Build the docker image with `docker build -t domire8/ros-pycharm-example .` (this will take a while the first
```

Comment:
It would be nice to have more emphasis on the steps, for example with
```suggestion
**Step 1:** Build the docker image with `docker build -t domire8/ros-pycharm-example .` (this will take a while the first
```